### ISSUE TITLE
normalize_sv_coverage failing - did not find inputs for sample

### ISSUE THREAD

**Issue by ameynert** [chapmanb/bcbio-nextgen#2253]
I'm trying to run some SV callers on a BAM file of WGS from a cell line. I suspect I've done something wrong with the configuration but haven't been able to spot it. All the file paths check out and the SM tag in the BAM file read group matches the description field in the YAML file.

Config YAML:
```
details:
- algorithm:
    aligner: false
    mark_duplicates: false
    remove_lcr: true
    variantcaller: false
    svcaller: [lumpy,manta,cnvkit]
    tools_on: [svplots]
  analysis: variant2
  description: Neo3
  files: /exports/igmm/eddie/bioinfsvice/ameynert/gilbert_neocentromere/samples/FD02163812/Neo3.bam
  genome_build: hg38
  metadata:
    batch: Neo
    phenotype: normal
upload:
  dir: /exports/igmm/eddie/bioinfsvice/ameynert/gilbert_neocentromere/upload/Neo3
```

bcbio output:
```
[2018-02-05T12:31Z] System YAML configuration: /gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/galaxy/bcbio_system.yaml
[2018-02-05T12:31Z] Resource requests: sambamba, samtools; memory: 8.00, 8.00; cores: 16, 16
[2018-02-05T12:31Z] Configuring 1 jobs to run, using 15 cores each with 125.4g of memory reserved for each job
[2018-02-05T12:31Z] Timing: organize samples
[2018-02-05T12:31Z] multiprocessing: organize_samples
[2018-02-05T12:31Z] Using input YAML configuration: /exports/igmm/eddie/bioinfsvice/ameynert/gilbert_neocentromere/config/Neo3.yaml
[2018-02-05T12:31Z] Checking sample YAML configuration: /exports/igmm/eddie/bioinfsvice/ameynert/gilbert_neocentromere/config/Neo3.yaml
[2018-02-05T12:31Z] Testing minimum versions of installed programs
[2018-02-05T12:31Z] Timing: alignment preparation
[2018-02-05T12:31Z] multiprocessing: prep_align_inputs
[2018-02-05T12:31Z] multiprocessing: disambiguate_split
[2018-02-05T12:31Z] Timing: alignment
[2018-02-05T12:31Z] multiprocessing: process_alignment
[2018-02-05T12:31Z] Index BAM file: Neo3.bam
[2018-02-05T14:17Z] Timing: callable regions
[2018-02-05T14:17Z] multiprocessing: prep_samples
[2018-02-05T14:17Z] multiprocessing: postprocess_alignment
[2018-02-05T14:17Z] Calculating coverage: Neo3 variant_regions
***** WARNING: File /gpfs/igmmfs01/eddie/bioinfsvice/ameynert/gilbert_neocentromere/work/Neo3/coverage/Neo3/Neo3-variant_regions.quantized.bed.gz has inconsistent naming convention for record:
HLA-A*01:01:01:01	0	2	NO_COVERAGE

***** WARNING: File /gpfs/igmmfs01/eddie/bioinfsvice/ameynert/gilbert_neocentromere/work/Neo3/coverage/Neo3/Neo3-variant_regions.quantized.bed.gz has inconsistent naming convention for record:
HLA-A*01:01:01:01	0	2	NO_COVERAGE

[2018-02-05T14:28Z] Neo3: Assigned coverage as 'genome' with 89.1% genome coverage and 0.0% offtarget coverage
[2018-02-05T14:28Z] Prepare cleaned BED file : Neo3
[2018-02-05T14:28Z] bgzip Neo3-variant_regions.quantized-vrsubset-callableblocks.bed
[2018-02-05T14:28Z] tabix index Neo3-variant_regions.quantized-vrsubset-callableblocks.bed.gz
[2018-02-05T14:28Z] Prepare merged BED file : Neo3
[2018-02-05T14:28Z] bgzip Neo3-variant_regions.quantized-vrsubset-callableblocks-merged.bed
[2018-02-05T14:28Z] tabix index Neo3-variant_regions.quantized-vrsubset-callableblocks-merged.bed.gz
[2018-02-05T14:28Z] multiprocessing: combine_sample_regions
[2018-02-05T14:28Z] Identified 348 parallel analysis blocks
Block sizes:
  min: 539
  5%: 1221.95
  25%: 19431.75
  median: 6304762.5
  75%: 17149833.75
  95%: 20676573.299999997
  99%: 26666734.20999998
  max: 40512843
Between block sizes:
  min: 251
  5%: 267.0
  25%: 295.0
  median: 327.0
  75%: 664.75
  95%: 2330.4999999999986
  99%: 16654.42999999995
  max: 50173

[2018-02-05T14:28Z] multiprocessing: calculate_sv_bins
[2018-02-05T14:28Z] CNVkit target
[2018-02-05T14:28Z] Detected file format: bed
[2018-02-05T14:28Z] Splitting large targets
[2018-02-05T14:29Z] Wrote /gpfs/igmmfs01/eddie/bioinfsvice/ameynert/gilbert_neocentromere/work/Neo3/bcbiotx/tmpQ6jEdt/Neo3-target.bed with 11557066 regions
[2018-02-05T14:29Z] CNVkit antitarget
[2018-02-05T14:29Z] Detected file format: bed
[2018-02-05T14:29Z] Detected file format: bed
[2018-02-05T14:29Z] Wrote /gpfs/igmmfs01/eddie/bioinfsvice/ameynert/gilbert_neocentromere/work/Neo3/bcbiotx/tmpR2CnVO/Neo3-antitarget.bed with 217 regions
[2018-02-05T14:29Z] multiprocessing: calculate_sv_coverage
[2018-02-05T14:29Z] Calculating coverage: Neo3 target
[2018-02-05T14:40Z] Calculating coverage: Neo3 antitarget
[2018-02-05T14:51Z] Annotate BED file with gene info
[2018-02-05T14:53Z] Annotate BED file with gene info
[2018-02-05T14:54Z] multiprocessing: normalize_sv_coverage
Traceback (most recent call last):
  File "/exports/igmm/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/tools/bin/bcbio_nextgen.py", line 241, in <module>
    main(**kwargs)
  File "/exports/igmm/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/tools/bin/bcbio_nextgen.py", line 46, in main
    run_main(**kwargs)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 43, in run_main
    fc_dir, run_info_yaml)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 87, in _run_toplevel
    for xs in pipeline(config, run_info_yaml, parallel, dirs, samples):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 134, in variant2pipeline
    samples = run_parallel("normalize_sv_coverage", [samples])
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 28, in run_parallel
    return run_multicore(fn, items, config, parallel=parallel)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 86, in run_multicore
    for data in joblib.Parallel(parallel["num_jobs"], batch_size=1)(joblib.delayed(fn)(x) for x in items):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 779, in __call__
    while self.dispatch_one_batch(iterator):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 625, in dispatch_one_batch
    self._dispatch(tasks)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 588, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 111, in apply_async
    result = ImmediateResult(func)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 332, in __init__
    self.results = batch()
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 131, in __call__
    return [func(*args, **kwargs) for func, args, kwargs in self.items]
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/utils.py", line 52, in wrapper
    return apply(f, *args, **kwargs)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 299, in normalize_sv_coverage
    return svregions.normalize_sv_coverage(*args)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/structural/regions.py", line 205, in normalize_sv_coverage
    assert inputs, "Did not find inputs for sample batch: %s" % (" ".join(dd.get_sample_name(x) for x in items))
AssertionError: Did not find inputs for sample batch: Neo3
```

**Comment by chapmanb** [comment]
Alison;
Thanks for the report and apologies about the issue. The issue here is that the logic to group things into inputs and background gets confused because you have a single batch (`Neo`) with one sample and it's labeled `normal` which is used as a background. Since it can't find a corresponding `tumor` sample it errors out. If you just remove `phenotype: normal` and restart it should work cleanly and process this as a single sample. Hope this gets everything working for your analysis.

**Comment by ameynert** [comment]
Brad, I made the suggested fix, and now get this error (and have now recalled that this is why I had phenotype: normal specified ...).

```
[2018-02-06T19:24Z] [2018-02-06T19:24:51.669650] [node2c24.ecdf.ed.ac.uk] [62361_1] [WorkflowRunner] Workflow successfully completed all tasks
[2018-02-06T19:24Z] [2018-02-06T19:24:51.669665] [node2c24.ecdf.ed.ac.uk] [62361_1] [WorkflowRunner] Elapsed time for full workflow: 9976 sec
Traceback (most recent call last):
  File "/exports/igmm/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/tools/bin/bcbio_nextgen.py", line 241, in <module>
    main(**kwargs)
  File "/exports/igmm/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/tools/bin/bcbio_nextgen.py", line 46, in main
    run_main(**kwargs)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 43, in run_main
    fc_dir, run_info_yaml)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 87, in _run_toplevel
    for xs in pipeline(config, run_info_yaml, parallel, dirs, samples):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 174, in variant2pipeline
    samples = structural.run(samples, run_parallel, "standard")
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 177, in run
    for xs in to_process.values()))
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 28, in run_parallel
    return run_multicore(fn, items, config, parallel=parallel)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 86, in run_multicore
    for data in joblib.Parallel(parallel["num_jobs"], batch_size=1)(joblib.delayed(fn)(x) for x in items):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 779, in __call__
    while self.dispatch_one_batch(iterator):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 625, in dispatch_one_batch
    self._dispatch(tasks)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 588, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 111, in apply_async
    result = ImmediateResult(func)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 332, in __init__
    self.results = batch()
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 131, in __call__
    return [func(*args, **kwargs) for func, args, kwargs in self.items]
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/utils.py", line 52, in wrapper
    return apply(f, *args, **kwargs)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 307, in detect_sv
    return structural.detect_sv(*args)
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 197, in detect_sv
    for svdata in caller_fn(items):
  File "/gpfs/igmmfs01/eddie/bioinfsvice/ameynert/software/bcbio-1.0.7/anaconda/lib/python2.7/site-packages/bcbio/structural/manta.py", line 31, in run
    if paired.normal_bam and "break-point-inspector" in dd.get_tools_on(data):
AttributeError: 'NoneType' object has no attribute 'normal_bam'
```

**Comment by chapmanb** [comment]
Alison -- sorry about that issue. That was a bug in 1.0.7 that we fixed for the 1.0.8 release. You could also manually apply the small patch if you don't want to upgrade:

https://github.com/chapmanb/bcbio-nextgen/commit/c12062263930202e473d6c69e8bb1aa9769970eb

Sorry again, hope that gets things running smoothly.

**Comment by ameynert** [comment]
Thanks, that's all working now.
